README.md
# More Locomotives

![banner](http://i.imgur.com/QlaZvZP.png)

A factorio mod which adds several more locomotives.

Forum: https://forums.factorio.com/viewtopic.php?f=93&t=21001

Screenshots: http://imgur.com/a/rrgEo

Download: https://github.com/Sebb767/more-locomotives/releases

## new locomotives 

### cargo locomotive

A cheaper, more efficient but slower diesel locomotive. 
- Is slower (max speed is about 90 km/h) and has better brakes.
- More fuel efficient (about 60%) than the stock locomotive.
- Is hardly affected by the wagon count.
- Has a bigger light cone and will activate it earlier.
- Cheaper to build: Only needs 12 engine units, 16 gears, 20 iron & 4 steel plates.

_Why should I want a slower locomotive?_ Well, it's actually not that slow. Its probably not the best choice for transporting things  over long distances, but its ideal for small transport within your factory. Also, you will survive being hit by it!

### high-speed locomotive

The complete opposite to the cargo locomotive. 
- Is extremly fast (max speed is about 360 km/h).
- Has pretty bad brakes and will need some time to get to max speed.
- 5 fuel slots - which you should _really_ fill. For comparison: A cargo locomotive will be powered by a single solid fuel block at max speed for about 1,45 min and a distance of about 1,9 km - this will run a high-speed locomotive for about 10 seconds! But this is enough for nearly 1 km, so if you can afford it, your ressources will arrive pretty quickly.

_Why should I want a faster locomotive?_ Let's be serious for a second.

## What's next?

* <strike>A faster but far more fuel consuming high-speed locomotive.</strike> Done!
* <strike>The cargo locomotive should not be impacted much by a high load.</strike> Done!
* Automatically dual-headed cargo locomotive.
* Speed upgrade for the cargo locomotive while keeping its efficiency.
* Electro locomotives!

## Credits

Thansk a lot to d3phoenix and his mod, [color coding](https://forums.factorio.com/viewtopic.php?f=93&t=13907). This mod is mostly based on his mod and he gave me the permission to use his code and graphics. Even more thanks go to hitzu, who made those graphics. 

Thanks to Florian "Fank" Kinder for the german translation.


changelog.md
# Changelog


## 0.3
- Added German Translation (thanks to Florian "Fank" Kinder)
- Removed unused tiles.lua, stubbed item-groups.lua
- Reduced the size of the finished zip
- Nerved the highspeed locomotive: It will now accelerate slower and take more time to brake.
- Cargo Loc speed incresed to about 86 km/h
- Added config option to let the cargo locomotive accelerate slower

prototypes/items.lua
require("prototypes.config")

t_items = {}


locomotive = util.table.deepcopy(data.raw["item"]["diesel-locomotive"])
locomotive.name = "cargo-locomotive"
locomotive.icon = MOD_NAME.."/graphics/locomotive/orange/icon.png"
locomotive.place_result = "cargo-locomotive"
table.insert(t_items,locomotive)

locomotive = util.table.deepcopy(data.raw["item"]["diesel-locomotive"])
locomotive.name = "hs-locomotive"
locomotive.icon = MOD_NAME.."/graphics/locomotive/red/icon.png"
locomotive.place_result = "hs-locomotive"
table.insert(t_items,locomotive)

locomotive = util.table.deepcopy(data.raw["item"]["diesel-locomotive"])
locomotive.name = "electro-locomotive"
locomotive.icon = MOD_NAME.."/graphics/locomotive/blue/icon.png"
locomotive.place_result = "electro-locomotive"
table.insert(t_items,locomotive)

data:extend(t_items)


prototypes/technologies.lua
require("prototypes.config")

t_technologies = {}


train = util.table.deepcopy(data.raw["technology"]["railway"])
train.name = "cargo-locomotive"
train.icon = MOD_NAME.."/graphics/locomotive/orange/icon.png"
train.prerequisites = {"railway"}
train.unit = {count=25, ingredients={{"science-pack-1",1},{"science-pack-2",1}}, time=20}
train.order = "c-g-ac"
train.effects = {
  { type="unlock-recipe", recipe="cargo-locomotive" }
}
table.insert(t_technologies, train)

train = util.table.deepcopy(data.raw["technology"]["railway"])
train.name = "hs-locomotive"
train.icon = MOD_NAME.."/graphics/locomotive/red/icon.png"
train.prerequisites = {"railway"}
train.unit = { count=75, ingredients={{"science-pack-1",2},{"science-pack-2",2},{"science-pack-3",1}}, time=30}
train.order = "c-g-ac"
train.effects = {
  { type="unlock-recipe", recipe="hs-locomotive" }
}
table.insert(t_technologies, train)

data:extend(t_technologies)


prototypes/entities.lua
require("prototypes.config")

new_trains = {}

-- Cargo Locomotive
cargo_loc = util.table.deepcopy(data.raw["locomotive"]["diesel-locomotive"])
cargo_loc.name = "cargo-locomotive"
cargo_loc.minable["result"] = cargo_loc.name
base_path = "/graphics/locomotive/orange"
cargo_loc.icon                  = MOD_NAME..base_path.."/icon.png"
cargo_loc.pictures.filenames[1] = MOD_NAME..base_path.."/01.png"
cargo_loc.pictures.filenames[2] = MOD_NAME..base_path.."/02.png"
cargo_loc.pictures.filenames[3] = MOD_NAME..base_path.."/03.png"
cargo_loc.pictures.filenames[4] = MOD_NAME..base_path.."/04.png"
cargo_loc.pictures.filenames[5] = MOD_NAME..base_path.."/05.png"
cargo_loc.pictures.filenames[6] = MOD_NAME..base_path.."/06.png"
cargo_loc.pictures.filenames[7] = MOD_NAME..base_path.."/07.png"
cargo_loc.pictures.filenames[8] = MOD_NAME..base_path.."/08.png"
cargo_loc.air_resistance = 0.0005
cargo_loc.max_health = 2500

cargo_loc.max_speed = 0.4
cargo_loc.max_power = "400kW"
if CARGOLOC_SLOW_ACCELERATION then
  cargo_loc.max_powers = "250kW"
end
cargo_loc.braking_force = 24
front_light_file = {
            filename = "__core__/graphics/light-cone.png",
            priority = "medium",
            scale = 3,
            width = 250,
            height = 250
          }
cargo_loc.front_light =
      {
        {
          type = "oriented",
          minimum_darkness = 0.2,
          picture = front_light_file,
          shift = {0.6, -20},
          size = 2,
          intensity = 0.6
        },
        {
          type = "oriented",
          minimum_darkness = 0.2,
          picture = front_light_file,
          shift = {-0.6, -20},
          size = 2,
          intensity = 0.6
        }
      }
cargo_loc.energy_source =
    {
      type = "burner",
      effectivity = 1.6,
      fuel_inventory_size = 3,
      smoke =
      {
        {
          name = "train-smoke",
          deviation = {0.3, 0.3},
          frequency = 80,
          position = {0, 0},
          starting_frame = 0,
          starting_frame_deviation = 60,
          height = 2,
          height_deviation = 0.5,
          starting_vertical_speed = 0.2,
          starting_vertical_speed_deviation = 0.1,
        }
      }
    }

table.insert(new_trains, cargo_loc)

-- high-speed locomotive
hs_loc = util.table.deepcopy(data.raw["locomotive"]["diesel-locomotive"])
hs_loc.name = "hs-locomotive"
hs_loc.minable["result"] = hs_loc.name
base_path = "/graphics/locomotive/red"
hs_loc.icon                  = MOD_NAME..base_path.."/icon.png"
hs_loc.pictures.filenames[1] = MOD_NAME..base_path.."/01.png"
hs_loc.pictures.filenames[2] = MOD_NAME..base_path.."/02.png"
hs_loc.pictures.filenames[3] = MOD_NAME..base_path.."/03.png"
hs_loc.pictures.filenames[4] = MOD_NAME..base_path.."/04.png"
hs_loc.pictures.filenames[5] = MOD_NAME..base_path.."/05.png"
hs_loc.pictures.filenames[6] = MOD_NAME..base_path.."/06.png"
hs_loc.pictures.filenames[7] = MOD_NAME..base_path.."/07.png"
hs_loc.pictures.filenames[8] = MOD_NAME..base_path.."/08.png"
hs_loc.braking_force = 15

hs_loc.max_speed = 1.8
hs_loc.max_power = "400kW"
hs_loc.energy_source =
    {
      type = "burner",
      effectivity = 0.4,
      fuel_inventory_size = 5,
      smoke =
      {
        {
          name = "train-smoke",
          deviation = {0.3, 0.3},
          frequency = 320,
          position = {0, 0},
          starting_frame = 0,
          starting_frame_deviation = 60,
          height = 2,
          height_deviation = 0.5,
          starting_vertical_speed = 0.2,
          starting_vertical_speed_deviation = 0.1,
        }
      }
    }



table.insert(new_trains, hs_loc)

-- electro locomotive
eloc = util.table.deepcopy(data.raw["locomotive"]["diesel-locomotive"])
eloc.name = "electro-locomotive"
eloc.minable["result"] = hs_loc.name
base_path = "/graphics/locomotive/blue"
eloc.icon                  = MOD_NAME..base_path.."/icon.png"
eloc.pictures.filenames[1] = MOD_NAME..base_path.."/01.png"
eloc.pictures.filenames[2] = MOD_NAME..base_path.."/02.png"
eloc.pictures.filenames[3] = MOD_NAME..base_path.."/03.png"
eloc.pictures.filenames[4] = MOD_NAME..base_path.."/04.png"
eloc.pictures.filenames[5] = MOD_NAME..base_path.."/05.png"
eloc.pictures.filenames[6] = MOD_NAME..base_path.."/06.png"
eloc.pictures.filenames[7] = MOD_NAME..base_path.."/07.png"
eloc.pictures.filenames[8] = MOD_NAME..base_path.."/08.png"
eloc.braking_force = 25

eloc.max_speed = 2.0
eloc.max_power = "1200kW"
eloc.air_resistance = 0.005
eloc.energy_source =
    {
      type = "burner",
      effectivity = 0.001,
      fuel_inventory_size = 0,
      smoke = {}
    }



table.insert(new_trains, eloc)

data:extend(new_trains)
